android-pdf-viewer/src/main/java/com/github/barteksc/pdfviewer/annotation/core/pdf/PdfUtil.kt

```diff
@@ -162,6 +166,38 @@ object PdfUtil {
         )
     }
 
+    private fun getExtractedLinkAnnotation(
+        annotation: PdfDictionary,
+        llx: Float,
+        lly: Float,
+        urx: Float,
+        ury: Float
+    ): Annotation {
+        val xBottomLeftPoint = llx
+        val yBottomLeftPoint = lly
+        val bottomLeftPoint = PointF(xBottomLeftPoint, yBottomLeftPoint)
+        val xTopRightPoint = urx
+        val yTopRightPoint = ury
+        val topRightPoint = PointF(xTopRightPoint, yTopRightPoint)
+
+        // from the extracted coordinates, calculate the rest
+        val linkAnnotationPoints = generateRectCoordinates(bottomLeftPoint, topRightPoint)
+
+        // extract URI from the action dictionary
+        val uriAction: PdfDictionary? = annotation.getAsDict(PdfName.A)
+        var uri: String? = null
+        if (uriAction != null) {
+            uri = uriAction.get(PdfName.URI)?.toString()
+        }
+
+        return Annotation(
+            AnnotationType.LINK.name,
+            linkAnnotationPoints,
```

Comment:
Is there a usage or need to have the points for this type as well?

Reply:
We need the coordinates so when we tap on annotation we can check it against the link annotations